### ISSUE TITLE
Create a database of features supported by the various targets

### ISSUE THREAD

**Issue by cmnrd** [icyphy/lingua-franca#237]
We are maintaining a rather heterogeneous project where we have multiple runtimes written in different languages and multiple code generators for those languages. Naturally the various targets differ in their approach due to differences in what the given language allows to describe, what is considered a good style in this language, and also just by personal preference of the programmer. I consider this a feature as it allows us to explore and compare different approaches. However, it becomes increasingly difficult to keep track of what is supported by which target. More importantly, it is hard to understand what the correct or expected behavior is (i.e. the one described by the formal reactor model). I think we currently see the C target as the reference implementation.  This target is actively developed and as it adds new features like the federated execution, it also often imposes adjustments in the "expected behavior". However, often these changes just happen without a good mechanism for keeping record of the changes and whether other targets adjust to them.

One example of such issues is the changed policy for scheduling logical actions. This is not implemented in C++ yet (see #236). While this is currently not implemented due to a lack of time from my side, I was also not aware of this important change for quite a while.

I am thinking that we need some kind of database, where we can keep track of (new) features and which targets support them. Also, we should carefully document any changes we make to the "expected behavior". I think we need some kind of version history for the "expected behavior" and also keep track which targets implement which version. Maybe we could just use a wiki page for this.

Are there any thoughts on this issue?

**Comment by edwardalee** [comment]
I view the [Language Specification](https://github.com/icyphy/lingua-franca/wiki/Language-Specification) wiki as the master specification.  It hasn’t been updated yet to reflect revisions to physical actions.

**Comment by Soroosh129** [comment]
I have updated that page to reflect the new behavior of physical actions. I have also updated [Physical Connections](https://github.com/icyphy/lingua-franca/wiki/Physical-Connections).

**Comment by cmnrd** [comment]
Yes, that is good to have! What I am trying to say is that we should try to additionally keep track of changes to this specification and how well our targets match the current specification. There are so many things that are going on. It is easy to lose track. Just by reading the specification it is neither obvious to us nor to any user to which degree the various targets actually match the specification or where they might differ.

**Comment by lhstrh** [comment]
One of the things I plan to do in January is build a website and move a lot of the wiki's content there. This will allow us to have more nicely formatted documentation. I think this would be a good place for publishing something like a feature or compatibility matrix. If, for the time being, we would maintain a wiki page that captures this information, I agree this would be helpful.

**Comment by lhstrh** [comment]
I created something [here](https://github.com/icyphy/lingua-franca/wiki/Support-Matrix) for the time being. We'll probably want to move this (along with most else that's currently on the wiki) into the user documentation once we have that up on our website.
